tests/microbenchmark/test_rw_tensix.cpp

```diff
@@ -0,0 +1,128 @@
+    std::uint32_t address = l1_mem::address_map::DATA_BUFFER_SPACE_BASE;
+
+    ankerl::nanobench::Bench bench;
+    for(auto& core :  device->get_virtual_soc_descriptors().at(0).workers) {
+        std::stringstream rname;
+        rname << "Read from device core (" << core.x << ", " << core.y << ")";
+        bench.title("Read 32 bytes").unit("reads").minEpochIterations(50).output(nullptr).run(rname.str(), [&] {
+            device->read_from_device(readback_vec, tt_cxy_pair(0, core), address, 0x20, "");
+        });
+        rname.clear();
+    }
+    bench.render(ankerl::nanobench::templates::csv(), results_csv);
+}
+
+void Write32BytesRandomAddr(std::shared_ptr<tt_SiliconDevice> device, std::ofstream& results_csv) {
+    std::vector<uint32_t> vector_to_write = {0, 1, 2, 3, 4, 5, 6, 7};
+    std::vector<uint32_t> readback_vec = {};
+    std::uint32_t address;
+
+    ankerl::nanobench::Bench bench;
+    for(auto& core :  device->get_virtual_soc_descriptors().at(0).workers) {
+        address = generate_random_address(1<<20); // between 0 and 1MB
+        std::stringstream wname;
+        wname << "Write to device core (" << core.x << ", " << core.y << ") @ address " << std::hex << address;
+        bench.title("Write 32 bytes random address").unit("writes").minEpochIterations(50).output(nullptr).run(wname.str(), [&] {
+            device->write_to_device(vector_to_write, tt_cxy_pair(0, core), address, "");
+        });
+        wname.clear();
+    }
+    bench.render(ankerl::nanobench::templates::csv(), results_csv);
+}
+
+void Read32BytesRandomAddr(std::shared_ptr<tt_SiliconDevice> device, std::ofstream& results_csv) {
+    std::vector<uint32_t> readback_vec = {};
+    std::uint32_t address;
+    auto &core = device->get_virtual_soc_descriptors().at(0).workers.at(0);
+
+    ankerl::nanobench::Bench bench;
+    for(int i = 0; i < 100; i++) {
+        address = generate_random_address(1<<20); // between 0 and 1MB
+        std::stringstream rname;
+        rname << "Read from device core (" << core.x << ", " << core.y << ") @ address " << std::hex << address;
+        bench.title("Read 32 bytes random address").unit("reads").minEpochIterations(50).output(nullptr).run(rname.str(), [&] {
+            device->read_from_device(readback_vec, tt_cxy_pair(0, core), address, 0x20, "");
+        });
+        rname.clear();
+    }
+    bench.render(ankerl::nanobench::templates::csv(), results_csv);
+}
+
+int main(){
+    std::ofstream results_csv("ubench_results.csv", std::ios_base::app);
+
+    auto get_static_tlb_index = [] (tt_xy_pair target) {
+        int flat_index = target.y * 10 + target.x;  // grid_size_x = 10 for GS/WH ????? something is wrong here
+        if (flat_index == 0) {
+            return -1;
+        }
+        return flat_index;
+    };
+    std::set<chip_id_t> target_devices = {0};
+    std::unordered_map<std::string, std::int32_t> dynamic_tlb_config = {{"SMALL_READ_WRITE_TLB", 157}}; // Use both static and dynamic TLBs here
```

Comment:
If all of the (x,y) cores and addresses in the benchmarks have a TLB configured below (`device->configure_tlb...`) then the "fallback" path (aim TLB 157 at the core in question and read/write through its window) won't happen -- UMD will see that it has an existing mapping and use it.

If you want to benchmark the fallback path, I think you need both:
* Use an address that isn't covered by a static TLB window (so UMD takes the fallback path)
* "SMALL_READ_WRITE_TLB" passed in to the read/write methods (so UMD knows which TLB index to use)

Regarding address coverage: note that although the TLB config code below is using a non-zero address within the core (`l1_mem::address_map::DATA_BUFFER_SPACE_BASE` is 0x33000 for me -- building for WH), the TLB window itself is still aimed at the bottom of the Tensix address space (0x0) and access through the window is offset by UMD.  So UMD would use the same TLB window to access e.g. `DATA_BUFFER_SPACE_BASE-4`.

Reply:
Yep this makes sense and I have a note for it on my side. I have ubenchmarked the fallback tlb path and it is much slower than the static path. 

Right now, these are just basic tests but I'm planning on expanding them/making them more general in the future.

Reply:
test for fallback/dynamic read/write path should be added now